api/src/main/java/com/graphhopper/util/shapes/BBox.java

```diff
@@ -161,6 +164,36 @@ public boolean contains(Shape s) {
         throw new UnsupportedOperationException("unsupported shape");
     }
 
+    private RectangleLineIntersector cachedIntersector;
+
+    RectangleLineIntersector getCachedRectangleLineIntersector() {
+        if (cachedIntersector == null)
+            cachedIntersector = new RectangleLineIntersector(toEnvelope(this));
+        return cachedIntersector;
+    }
+
+    @Override
+    public boolean intersects(PointList pointList) {
+        int len = pointList.getSize();
+        if (len == 0)
+            throw new IllegalArgumentException("PointList must not be empty");
+
+        double lat = pointList.getLatitude(0);
+        double lon = pointList.getLongitude(0);
+        if (len == 1)
+            return contains(lat, lon);
+
+        Coordinate coords = new Coordinate(lon, lat);
+        RectangleLineIntersector intersector = getCachedRectangleLineIntersector();
```

Comment:
I think this will fail when this method is called after this object was updated with update(). BBox is mutable, so I think we can't cache the RectangleLineIntersector.

Reply:
Yeah, will see if this is critical for speed and if there is a different possibility.

Reply:
Have moved the cached RectangleLineIntersector upwards to GraphEdgeIdFinder.parseBlockArea

Reply:
Have moved the cached RectangleLineIntersector upwards to GraphEdgeIdFinder.parseBlockArea